src/passes/LogExecution.cpp

```diff
@@ -81,14 +106,40 @@ struct LogExecution : public WalkerPass<PostWalker<LogExecution>> {
 
     import->base = LOGGER;
     curr->addFunction(std::move(import));
+
+    // Reserve all function indices up front for the function names. This is
+    // so that the logged ordinal numbers will match up with the function ordinals.
+    int idx = 0;
+    for (auto& func : curr->functions) {
+       if (func->imported())
+         ++idx;
+    }
+
+    for (auto& func : curr->functions) {
+      if (func->imported())
+        continue;
+
+      Index currentFunctionIndex = (Index)stringToIndex(func->name.toString().c_str());
```

Comment:
```suggestion
      Index currentFunctionIndex = stringToIndex(func->name.toString().c_str());
```

Also, why try to convert function names to indices at all? It would be more robust to avoid trying to interpret the names so that all function names can be treated uniformly.